### PR TITLE
support negative indeces in list slices

### PR DESCRIPTION
**Pull request by dbyron0**
so e.g.

{{ (split_me|split('-'))[0:-2] }}

works

### CODE REVIEW

File: src/main/java/com/hubspot/jinjava/el/ext/AstRangeBracket.java [hunk 1]
```diff
@@ -72,6 +72,17 @@ public Object eval(Bindings bindings, ELContext context) {
     PyList result = new PyList(new ArrayList<>());
     int index = 0;
 
+    // Handle negative indeces.
+    if ((startNum < 0) || (endNum < 0)) {
+       int size = Iterables.size(baseItr);
```
Comment (dbyron0) [inline src/main/java/com/hubspot/jinjava/el/ext/AstRangeBracket.java]: If it's cheaper enough to use array length here I can special case that..

Comment (mattcoley) [reply src/main/java/com/hubspot/jinjava/el/ext/AstRangeBracket.java]: `Iterables.size()` has some smart logic to determine the fastest way to calculate the size depending on the underlying iterable implementation IIRC.

Comment (dbyron0) [reply src/main/java/com/hubspot/jinjava/el/ext/AstRangeBracket.java]: Sorry, ignore me.  Arrays turn into lists which are cheap for calculating size...

**Comment by dbyron0** [comment]
Sure would love to get this in.  Any more I can do?

**Pull request merged**